### PR TITLE
[GH-2091] WKB, WKT reader & writer

### LINKED ISSUE #2091: [GH-2091] WKB, WKT reader & writer
- PR: https://github.com/apache/sedona/pull/2090 
- WKB 
- WKT
- new extend type of : SinglePoint, SinglePolyline, Multipolygon

### PR DESCRIPTION
**Pull request by ZhuochengShang**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest/community/develop/)

## Is this PR related to a ticket?

- Yes, and the PR name follows the format `[GH-XXX] my subject`. Closes #2091 

## What changes were proposed in this PR?
- WKB 
- WKT
- new extend type of : SinglePoint, SinglePolyline, Multipolygon

## How was this patch tested?

Tested parsing and writing of WKB/WKT using all types of geography objects with various internal structures.

## Did this PR include necessary documentation updates?

- No, this PR does not affect any public API so no need to change the documentation.

### CODE REVIEW

**Review by Kontinuation — COMMENTED** [review]
We have added `getCoordinateSequence` methods to geography types for easier reusing JTS WKB/WKT writer, and the geography readers read coordinates into JTS CoordinateSequence internally and then convert to S2 objects. This introduces extra data copying but let us do minimal modification based on JTS's code.

Now I am less concerned about the performance implications of this approach, because I found that WKT/WKB reading/writing involves conversion between R3 vectors and lat/lng, which involves calling multiple trigonometric functions (see [S2LatLng.toPoint](https://github.com/google/s2-geometry-library-java/blob/7d860272c40184b71a7c456583763b142d4118f9/library/src/com/google/common/geometry/S2LatLng.java#L263-L275) and [constructor of S2LatLng](https://github.com/google/s2-geometry-library-java/blob/7d860272c40184b71a7c456583763b142d4118f9/library/src/com/google/common/geometry/S2LatLng.java#L171-L180)). This overhead out weights any effort we make to optimize the parsing/writing of WKB/WKT themselves.

The remaining work of this PR should be

* Add more test cases for WKB reader/writer. Including empty geometries, multi-linestring, multi-polygon, and geography collection.
* Remove unused code

File: common/src/main/java/org/apache/sedona/common/S2Geography/WKBWriter.java [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+    writeGeometryType(geometryType, pt, os);
+    if (pt.numShapes() == 0) {
+      writeNaNs(outputDimension, os);
+    } else {
+      S2Point p = pt.getPoints().get(0);
+      S2LatLng ll = new S2LatLng(p);
+
+      // round to 6 decimal places //TODO: 16 decimal
+      double lon = Math.rint(ll.lngDegrees() * 1e16) / 1e16;
+      double lat = Math.rint(ll.latDegrees() * 1e16) / 1e16;
```
Comment (Kontinuation) [inline common/src/main/java/org/apache/sedona/common/S2Geography/WKBWriter.java]: The WKBWriter does not have take care about precision model. I believe that the handling of maximum significant digits only happens in WKT writer?

File: common/src/test/java/org/apache/sedona/common/S2Geography/WKTReadWriterTest.java [hunk 2]
```diff
@@ -117,4 +113,18 @@ public void collection_roundTrip() throws ParseException {
...
+    assertEquals(
+        point, writeWithPrecision(reader.read(point), new PrecisionModel(PrecisionModel.FIXED)));
+    String line = "LINESTRING EMPTY";
+    assertEquals(
+        line, writeWithPrecision(reader.read(line), new PrecisionModel(PrecisionModel.FIXED)));
+    String polygon = "POLYGON EMPTY";
+    assertEquals(
+        polygon,
+        writeWithPrecision(reader.read(polygon), new PrecisionModel(PrecisionModel.FIXED)));
+  }
```
Comment (Kontinuation) [inline common/src/test/java/org/apache/sedona/common/S2Geography/WKTReadWriterTest.java]: Can we add tests for `MULTIPOINT EMPTY`, `MULTILINESTRING EMPTY`, `MULTIPOLYGON EMPTY` and `GEOMETRYCOLLECTION EMPTY` for completeness?

File: common/src/test/java/org/apache/sedona/common/S2Geography/TestHelper.java [hunk 3]
```diff
@@ -18,18 +18,23 @@
  */
 package org.apache.sedona.common.S2Geography;
 
+import static org.apache.sedona.common.S2Geography.Accessors.S2_isEmpty;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
 import com.esotericsoftware.kryo.io.UnsafeInput;
 import com.google.common.geometry.*;
+import java.awt.*;
```
Comment (Kontinuation) [inline common/src/test/java/org/apache/sedona/common/S2Geography/TestHelper.java]: I believe this is not needed.

**Review by Kontinuation — APPROVED** [review]
I think it is good to merge now.

**Review by jiayuasu — APPROVED** [review]
